### PR TITLE
Cmd encode decode fixes

### PR DESCRIPTION
**Pull request by plorenz**

### CODE REVIEW

File: controller/model/posture_check_manager.go [hunk 1]
```diff
@@ -144,7 +145,10 @@ func (self *PostureCheckManager) Marshall(entity *PostureCheck) ([]byte, error)
 		RoleAttributes: entity.RoleAttributes,
 	}
 
-	entity.SubType.fillProtobuf(msg)
+	if entity.SubType != nil {
```
Comment (andrewpmartinez) [inline controller/model/posture_check_manager.go:148]: Interesting, subtype should never be nil.

Comment (andrewpmartinez) [reply controller/model/posture_check_manager.go:148]: Unless this is to account for a partial update? Say a name change only?

File: pb/edge_cmd_pb/impl.go [hunk 2]
```diff
@@ -83,78 +87,159 @@ func DecodeTags(tags map[string]*TagValue) map[string]interface{} {
...
+		encoded, err := EncodeJsonValue(v)
+		if err != nil {
+			return nil, err
+		}
+		result[k] = encoded
+	}
+	return &JsonMap{Value: result}, nil
+}
+
+func EncodeJsonValue(v interface{}) (*JsonValue, error) {
```
Comment (andrewpmartinez) [inline pb/edge_cmd_pb/impl.go:106]: I don't like this. I don't understand why we need this.

File: controller/model/posture_check_manager.go (hunk 1 above)
Comment (plorenz) [reply controller/model/posture_check_manager.go:148]: Yes, I think it happened in one of the acceptance tests with a PATCH

File: pb/edge_cmd_pb/impl.go (hunk 2 above)
Comment (plorenz) [reply pb/edge_cmd_pb/impl.go:106]: It's not used any more and I was going to remove it in a follow up, but I can remove it now

Comment (plorenz) [reply pb/edge_cmd_pb/impl.go]: removed

**Review by andrewpmartinez — APPROVED** [review]

**Pull request merged**